### PR TITLE
Add fix to cover an edge case in tweet unfavorite path

### PR DESCRIPTION
**Pull request by guimingTang**
Background:
In the previous pull request I added the functionality to index unfavorite edges, so they can be used to "remove" the corresponding favorite edges. The core idea is that, for each tweet (nodeInfo), check both favorite edges and unfavorite edges, find intersection, remove the intersection, and lastly remove the remainder unfavorite edges. The end result will appear as if unfavorited edges are removed.

Problem:
I neglected a corner case where favorite edges == null and unfavorite edges != null. This allowed the call path to skip the entire unfavorite check, and unfav social proofs will not be checked or removed. This pull request covers that corner case, and added tests to cover it.

Also I moved the fave and unfav edge type definitions to RecommendationRequest.java, where most of the constants are already defined, to reduce duplication.

### CODE REVIEW

File: graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/tweet/TopSecondDegreeByCountTweetRecsGenerator.java [hunk 1]
```diff
@@ -60,35 +60,40 @@ private static boolean removeUnfavoriteSocialProofs(NodeInfo nodeInfo) {
     }
 
     SmallArrayBasedLongToDoubleMap[] socialProofs = nodeInfo.getSocialProofs();
-    SmallArrayBasedLongToDoubleMap unfavSocialProofs = socialProofs[UnfavoriteSocialProofType];
-    SmallArrayBasedLongToDoubleMap favSocialProofs = socialProofs[FavoriteSocialProofType];
+    SmallArrayBasedLongToDoubleMap unfavSocialProofs = socialProofs[UNFAVORITE_SOCIAL_PROOF_TYPE];
+    SmallArrayBasedLongToDoubleMap favSocialProofs = socialProofs[FAVORITE_SOCIAL_PROOF_TYPE];
 
-    if (unfavSocialProofs == null || favSocialProofs == null) {
+    if (unfavSocialProofs == null) {
```
Comment (jerryjiangabc) [inline graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/tweet/TopSecondDegreeByCountTweetRecsGenerator.java]: could you add more comments on why this change is necessary?
